Ui

cehsu opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/sfbrigade/sfsbook/103)
<!-- Reviewable:end -->

rjkroege commented:
Reviewed 23 of 24 files at r1.
Review status: 23 of 24 files reviewed at latest revision, 21 unresolved discussions.

---

*[server/adduser.go, line 35 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdH22HnxVNg2_4xhXaJ:-KdH22HnxVNg2_4xhXaK:b-vs465i) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/server/adduser.go#L35)):*
> ```Go
> func (gs *addUser) ender(w http.ResponseWriter, req *http.Request, listusersresult interface{}) {
> 	sn := req.URL.Path
> 	templates := []string{sn, "/head.html", "/header.html","/searchbar.html", "/footer.html"}
> ```

I am not convinced that adduser should have the search bar? It seemed odd and distracting from the admin pages?

---

*[server/listusers.go, line 50 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdH3JDgt8enUEKt2GPL:-KdH3JDgt8enUEKt2GPM:benpsw9) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/server/listusers.go#L50)):*
> ```Go
> func (gs *listUsers) ender(w http.ResponseWriter, req *http.Request, listusersresult interface{}) {
> 	sn := req.URL.Path
> 	templates := []string{sn, "/head.html", "/header.html", "/searchbar.html", "/footer.html"}
> ```

Same comment as above.

---

*[server/listusers.go, line 161 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdH3dpbVVbzxYHQM50u:-KdH3dpbVVbzxYHQM50v:brjquwj) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/server/listusers.go#L161)):*
> ```Go
> 			action = _DELETEUSERS
> 		case k == "userquery":
> 			userquery, err := getValidatedString("userquery", req.Form)
> ```

This is a merge conflict. I think it will break the tests. Perhaps you could rebase?

---

*[server/login.go, line 171 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdH3xg4e8RfNXyLjeZz:-KdH3xg4e8RfNXyLje_-:b-ykp4eo) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/server/login.go#L171)):*
> ```Go
> end:
> 
> 	templates := []string{sn, "/head.html", "/header.html", "/searchbar.html", "/footer.html"}
> ```

As above: search maybe doesn't belong in the login page? And I'd like to keep this page simple until I've inserted the oauth integration.

---

*[server/templating.go, line 39 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdH5rl9tjHAxh3YywWB:-KdH5rl9tjHAxh3YywWC:b324r9s) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/server/templating.go#L39)):*
> ```Go
> 	// in the database.
> 	results := gs.generator.ForRequest(req)
> 	templates := []string{sn, "/head.html", "/header.html", "/searchbar.html", "/footer.html"}
> ```

I forget which pages use `templating.go`. Maybe this file is misnamed. Is search bar right here?

---

*[server/usermgt.go, line 148 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdH6IdC42L3yUF1uVLt:-KdH6IdDqTHImWndrhF5:boqi2fy) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/server/usermgt.go#L148)):*
> ```Go
> 
> 	sn := req.URL.Path
> 	templates := []string{sn, "/head.html", "/header.html", "/searchbar.html", "/footer.html"}
> ```

Per comment above... no search bar here in my opinion. What do you think?

---

*[server/usermgt_test.go, line 30 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdH6V1ytvKjomirUeyB:-KdH6V1ytvKjomirUeyC:b-vkk7uf) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/server/usermgt_test.go#L30)):*
> ```Go
> 
> 	Resources = map[string]string{
> 		"/usermgt/changepasswd.html": embeddedResourceForUsermgt,
> ```

That we require this is a sign of a problem in the code. It would be easier/better perhaps if the code that looks this up was more tolerant of missing additional template files?

---

*[site/footer.html, line 6 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdH7HrJvgGH12EUGZ9a:-KdH7HrJvgGH12EUGZ9b:bitm4ce) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/site/footer.html#L6)):*
> ```HTML
>   <div class="categories"><a href="https://www.sfwar.org/programs.html">Activities</a></div>
>   <div class="categories"><a href="https://www.sfwar.org/involved.html">Get Involved</a></div>
>   <div class="categories"><a href="https://www.givedirect.org/donate/?cid=11469">Donate</a></div>
> ```

Remove this maybe? I don't know.

---

*[site/head.html, line 8 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdHPfmUG9WK2faTMFo7:-KdHPfmVFIJTitqvqABY:b-4f56rh) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/site/head.html#L8)):*
> ```HTML
>   <!--[if lt IE 9]><script type="text/javascript" src="/js/ie8-html5.js"></script><![endif]-->
>   <script type="text/javascript" src="/js/main.js"></script>
>   <link href="https://fonts.googleapis.com/css?family=Source+Sans+Pro" rel="stylesheet">
> ```

Could we suppress webfonts from mobile devices? 

---

*[site/login.html, line 7 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdHQqBF4bwQYlV9Nt7Y:-KdHQqBF4bwQYlV9Nt7Z:brif0nr) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/site/login.html#L7)):*
> ```HTML
>       {{template "header" .}}
>       <!-- need to add a status bar -->
>       {{template "searchbar" .}}
> ```

per above, I'd remove this.

---

*[site/searchbar.html, line 12 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdHRKV1B0S_F5n0FeCr:-KdHRKV1B0S_F5n0FeCs:bwc9lbu) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/site/searchbar.html#L12)):*
> ```HTML
>         <li class='category' role='menuitem' tabindex='-1' aria-haspopup='true'>
>           Locations<span class="chevron right"></span>
>           <ul class='category-dropdown category-hidden' role='menu'>
> ```

we should build this from the data in a subsequent set of changes. :-)

---

*[site/css/style.css, line 1 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdHR_-IwodRvmQMa0BQ:-KdHR_-IwodRvmQMa0BR:b-4pe6zz) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/site/css/style.css#L1)):*
> ```CSS
> /* Table of Contents */
> ```

I don't really know CSS very well so am relying on you here. :-)

---

*[site/js/helpers.js, line 17 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdI5Qh9Rada4G7DoiRw:-KdI5QhAMUACnKSchHx1:b-2jw7cy) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/site/js/helpers.js#L17)):*
> ```JavaScript
>  * @return {object} The selected node.
>  */
> function query(identifier) {
> ```

I'd make this qs or some such? And document.getElementById could also have a short form? But you can do this in a subsequent CL if you want. Or ignore me.

---

*[site/js/helpers.js, line 41 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdI5fmmL45dvg0irCRm:-KdI5fmmL45dvg0irCRn:b6u6xxq) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/site/js/helpers.js#L41)):*
> ```JavaScript
> }
> 
> /** Delete the session cookie and reload the page in the unauthed state. */
> ```

fyi: There is server side code to write to expire cookies. The client can't be trusted.

---

*[site/js/helpers.js, line 53 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdI7VYnBNjpEgrFo2ad:-KdI7VYoV43ZTnme4gmS:boqg91u) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/site/js/helpers.js#L53)):*
> ```JavaScript
>  * @param {object} node The selected navigation node.
>  */
> function userNav(node) {
> ```

You are 

---

*[site/js/helpers.js, line 58 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdI6r2pELVBgyHehH6n:-KdI6r2pELVBgyHehH6o:b-bqo0ri) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/site/js/helpers.js#L58)):*
> ```JavaScript
>     case keys.down:
>       if (theTarget.classList[0].indexOf('user-menu-item') > -1) {
>         var siblings = Array.prototype.slice.call(document.querySelectorAll('.user-menu-item'));
> ```

Seems like there is opportunity for additional code cleanup here? There is a query function? Maybe create and return the JavaScript array? Or would that always copy the element list? I forget what the storage implications are of preserving refs from JS heap to DOM. 

Also: it might not be a good idea performance to call `querySelectorAll` on the event handling critical path. It would be better to cache the list of menu items in advance? Or do we think that we will be editing the list of choices?

---

*[site/js/helpers.js, line 125 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdI8i9TtBci2NbjsDtV:-KdI8i9TtBci2NbjsDtW:b-aflr8e) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/site/js/helpers.js#L125)):*
> ```JavaScript
>       break;
> 
>     
> ```

what is this?

---

*[site/js/helpers.js, line 135 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdI8wGAyqizUAMV7bnN:-KdI8wGBqVHp0uYWoFHY:bko1jw4) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/site/js/helpers.js#L135)):*
> ```JavaScript
>     case keys.up:
>       diff = (event.which === 40) ? 1 : -1;
>       if (theTarget.classList[0].indexOf('category-option') > -1) {
> ```

this is O(n) in the size of classList. How big is classList? As per mumble above, caching would be helpful?

---

*[site/js/main.js, line 8 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdI9LIBW4U0jxN_Kj7Q:-KdI9LIBW4U0jxN_Kj7R:b-humoga) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/site/js/main.js#L8)):*
> ```JavaScript
>  */
> function addEventListener(el, eventName, handler) {
>   eventName.preventDefault ? eventName.preventDefault() :
> ```

can you explain how this should work? Maybe in a comment?

---

*[site/usermgt/adduser.html, line 13 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdI9eOVMKP3I3PCbp-M:-KdI9eOVMKP3I3PCbp-N:b1vxgl5) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/site/usermgt/adduser.html#L13)):*
> ```HTML
>       <p class="notification">{{.Username}} successfully added.</p>
>       {{else if .Usernotadded}}
>         {{.ReasonForFailure}}
> ```

looks like a merge conflict of some kind?

---

*[site/usermgt/listusers.html, line 46 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdI9sk1tZ4g_2JIL6AB:-KdI9sk1tZ4g_2JIL6AC:bq8p8lx) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/site/usermgt/listusers.html#L46)):*
> ```HTML
>                 <option value='norole'>Clear role</option>
>               </select>
>             </div>            
> ```

 spaces I think... 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103)*
<!-- Sent from Reviewable.io -->

cehsu commented:
Review status: 23 of 24 files reviewed at latest revision, 19 unresolved discussions.

---

*[server/adduser.go, line 35 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdH22HnxVNg2_4xhXaJ:-KdX6M20CRFI3rsYwJdX:btp5ybc) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/server/adduser.go#L35)):*
<details><summary><i>Previously, rjkroege (Robert Kroeger) wrote…</i></summary><blockquote>

I am not convinced that adduser should have the search bar? It seemed odd and distracting from the admin pages?
</blockquote></details>

Good point. removed.

---

*[server/listusers.go, line 50 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdH3JDgt8enUEKt2GPL:-KdXErwajqWRoWjv-sHN:bmnd3zm) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/server/listusers.go#L50)):*
<details><summary><i>Previously, rjkroege (Robert Kroeger) wrote…</i></summary><blockquote>

Same comment as above.
</blockquote></details>

Good point. Removed.

---

*[server/listusers.go, line 161 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdH3dpbVVbzxYHQM50u:-KdYr6HwbCBHIcLkXOp3:bi3zqy4) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/server/listusers.go#L161)):*
<details><summary><i>Previously, rjkroege (Robert Kroeger) wrote…</i></summary><blockquote>

This is a merge conflict. I think it will break the tests. Perhaps you could rebase?
</blockquote></details>

Good observation. Done.

---

*[server/login.go, line 171 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdH3xg4e8RfNXyLjeZz:-KdXFPstXDO07g-ZvQ90:bmnd3zm) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/server/login.go#L171)):*
<details><summary><i>Previously, rjkroege (Robert Kroeger) wrote…</i></summary><blockquote>

As above: search maybe doesn't belong in the login page? And I'd like to keep this page simple until I've inserted the oauth integration.
</blockquote></details>

Good point. Removed.

---

*[server/templating.go, line 39 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdH5rl9tjHAxh3YywWB:-KdYzwCmuwhAQXaPgRJ3:b-6cozyc) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/server/templating.go#L39)):*
<details><summary><i>Previously, rjkroege (Robert Kroeger) wrote…</i></summary><blockquote>

I forget which pages use `templating.go`. Maybe this file is misnamed. Is search bar right here?
</blockquote></details>

This is used when navigation to index.html, searchbar makes sense here, the name "templating.go" should be rethought, perhaps the file should be split into two?

---

*[server/usermgt.go, line 148 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdH6IdC42L3yUF1uVLt:-KdXFb5ucaLFdrZN99Gd:bs2vhuc) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/server/usermgt.go#L148)):*
<details><summary><i>Previously, rjkroege (Robert Kroeger) wrote…</i></summary><blockquote>

Per comment above... no search bar here in my opinion. What do you think?
</blockquote></details>

Good point, removed.

---

*[server/usermgt_test.go, line 30 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdH6V1ytvKjomirUeyB:-KdXkVCHHfC6Qn56l1J9:b-4p63sm) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/server/usermgt_test.go#L30)):*
<details><summary><i>Previously, rjkroege (Robert Kroeger) wrote…</i></summary><blockquote>

That we require this is a sign of a problem in the code. It would be easier/better perhaps if the code that looks this up was more tolerant of missing additional template files?
</blockquote></details>

I don't imagine a use case at present for including the partials in the templates but not providing them serverside. Maybe we should change this when that is necessary?

---

*[site/footer.html, line 6 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdH7HrJvgGH12EUGZ9a:-KdYYp42y3-021jxiAOU:b-sgpiwt) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/site/footer.html#L6)):*
<details><summary><i>Previously, rjkroege (Robert Kroeger) wrote…</i></summary><blockquote>

Remove this maybe? I don't know.
</blockquote></details>

Perhaps we should include this for now, for the purposes of eliciting user feedback?

---

*[site/head.html, line 8 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdHPfmUG9WK2faTMFo7:-KdXL4Q7WTeFO1KRoHNS:bhfo72f) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/site/head.html#L8)):*
<details><summary><i>Previously, rjkroege (Robert Kroeger) wrote…</i></summary><blockquote>

Could we suppress webfonts from mobile devices? 
</blockquote></details>

Good point, media query now provided.

---

*[site/login.html, line 7 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdHQqBF4bwQYlV9Nt7Y:-KdXFsyaEw8CYxp1Qcq6:bs2vhuc) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/site/login.html#L7)):*
<details><summary><i>Previously, rjkroege (Robert Kroeger) wrote…</i></summary><blockquote>

per above, I'd remove this.
</blockquote></details>

Good point, removed.

---

*[site/searchbar.html, line 12 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdHRKV1B0S_F5n0FeCr:-KdYrC0p5gULScXOItlp:b-c7xa45) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/site/searchbar.html#L12)):*
<details><summary><i>Previously, rjkroege (Robert Kroeger) wrote…</i></summary><blockquote>

we should build this from the data in a subsequent set of changes. :-)
</blockquote></details>

Added as an issue.

---

*[site/css/style.css, line 1 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdHR_-IwodRvmQMa0BQ:-KdXkLVIMzxFM8iTSkZ3:b-gjviuv) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/site/css/style.css#L1)):*
<details><summary><i>Previously, rjkroege (Robert Kroeger) wrote…</i></summary><blockquote>

I don't really know CSS very well so am relying on you here. :-)
</blockquote></details>

Excellent point about the descendent selectors, etc. Will optimize CSS in following CL.

---

*[site/js/helpers.js, line 53 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdI7VYnBNjpEgrFo2ad:-KdYrgTDSYHZMcWzJr5P:b3kre) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/site/js/helpers.js#L53)):*
<details><summary><i>Previously, rjkroege (Robert Kroeger) wrote…</i></summary><blockquote>

You are 
</blockquote></details>

?

---

*[site/js/helpers.js, line 58 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdI6r2pELVBgyHehH6n:-KdXa-ImF2RIG90cLQFX:b1lohmc) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/site/js/helpers.js#L58)):*
<details><summary><i>Previously, rjkroege (Robert Kroeger) wrote…</i></summary><blockquote>

Seems like there is opportunity for additional code cleanup here? There is a query function? Maybe create and return the JavaScript array? Or would that always copy the element list? I forget what the storage implications are of preserving refs from JS heap to DOM. 

Also: it might not be a good idea performance to call `querySelectorAll` on the event handling critical path. It would be better to cache the list of menu items in advance? Or do we think that we will be editing the list of choices?
</blockquote></details>

Excellent point, additional shortcuts created. Will add caching as a todo for the next CL.

---

*[site/js/helpers.js, line 125 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdI8i9TtBci2NbjsDtV:-KdXa5UYwRYrGNN3-Xwq:ba3lv4k) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/site/js/helpers.js#L125)):*
<details><summary><i>Previously, rjkroege (Robert Kroeger) wrote…</i></summary><blockquote>

what is this?
</blockquote></details>

Good catch, unnecessary whitespace removed.

---

*[site/js/helpers.js, line 135 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdI8wGAyqizUAMV7bnN:-KdXaDtHgp5dBebNajD1:bdifjd3) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/site/js/helpers.js#L135)):*
<details><summary><i>Previously, rjkroege (Robert Kroeger) wrote…</i></summary><blockquote>

this is O(n) in the size of classList. How big is classList? As per mumble above, caching would be helpful?
</blockquote></details>

Excellent point, made constant time complexity.

---

*[site/js/main.js, line 8 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdI9LIBW4U0jxN_Kj7Q:-KdYzmHFVyFaYKq4XNq4:b78z13e) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/site/js/main.js#L8)):*
<details><summary><i>Previously, rjkroege (Robert Kroeger) wrote…</i></summary><blockquote>

can you explain how this should work? Maybe in a comment?
</blockquote></details>

I've added additional description, in the JSDOC block, please let me know if additional explanation would be helpful.

---

*[site/usermgt/adduser.html, line 13 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdI9eOVMKP3I3PCbp-M:-KdYqi-j22xfBzV82wN-:b9ayhh4) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/site/usermgt/adduser.html#L13)):*
<details><summary><i>Previously, rjkroege (Robert Kroeger) wrote…</i></summary><blockquote>

looks like a merge conflict of some kind?
</blockquote></details>

Including Diagnosticmessage instead of ReasonForFailure results in the following message on failed user addition: 
"Can't execute template template: htmlbase:13:10: executing "htmlbase" at <.Diagnosticmessage>: can't evaluate field Diagnosticmessage in type *server.addUserResult"

---

*[site/usermgt/listusers.html, line 46 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdI9sk1tZ4g_2JIL6AB:-KdXjf-3pOc2NhA_Pc6j:b-yz1nik) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/site/usermgt/listusers.html#L46)):*
<details><summary><i>Previously, rjkroege (Robert Kroeger) wrote…</i></summary><blockquote>

 spaces I think... 
</blockquote></details>

good catch, unnecessary whitespace removed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103)*
<!-- Sent from Reviewable.io -->

cehsu commented:
Review status: 14 of 24 files reviewed at latest revision, 18 unresolved discussions.

---

*[server/adduser.go, line 35 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdH22HnxVNg2_4xhXaJ:-Kdd24ZyWpUYTdsuGVsw:b-896fix) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/server/adduser.go#L35)):*
<details><summary><i>Previously, cehsu (Claire Hsu) wrote…</i></summary><blockquote>

Good point. removed.
</blockquote></details>

Done.

---

*[server/listusers.go, line 50 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdH3JDgt8enUEKt2GPL:-Kdd263Rw7cSZqnXv3z9:b-896fix) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/server/listusers.go#L50)):*
<details><summary><i>Previously, cehsu (Claire Hsu) wrote…</i></summary><blockquote>

Good point. Removed.
</blockquote></details>

Done.

---

*[server/listusers.go, line 161 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdH3dpbVVbzxYHQM50u:-Kdd26zRmUOFY48jlLRY:b-896fix) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/server/listusers.go#L161)):*
<details><summary><i>Previously, cehsu (Claire Hsu) wrote…</i></summary><blockquote>

Good observation. Done.
</blockquote></details>

Done.

---

*[server/login.go, line 171 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdH3xg4e8RfNXyLjeZz:-Kdd2865WBIRrtUsaQQJ:b-896fix) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/server/login.go#L171)):*
<details><summary><i>Previously, cehsu (Claire Hsu) wrote…</i></summary><blockquote>

Good point. Removed.
</blockquote></details>

Done.

---

*[server/templating.go, line 39 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdH5rl9tjHAxh3YywWB:-Kdd28mjOGaVtbeTr8_r:b-896fix) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/server/templating.go#L39)):*
<details><summary><i>Previously, cehsu (Claire Hsu) wrote…</i></summary><blockquote>

This is used when navigation to index.html, searchbar makes sense here, the name "templating.go" should be rethought, perhaps the file should be split into two?
</blockquote></details>

Done.

---

*[server/usermgt.go, line 148 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdH6IdC42L3yUF1uVLt:-Kdd29Qgn0cyJNg3-Uv8:b-896fix) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/server/usermgt.go#L148)):*
<details><summary><i>Previously, cehsu (Claire Hsu) wrote…</i></summary><blockquote>

Good point, removed.
</blockquote></details>

Done.

---

*[server/usermgt_test.go, line 30 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdH6V1ytvKjomirUeyB:-Kdd2A0jynoj5zNnbiyG:b-896fix) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/server/usermgt_test.go#L30)):*
<details><summary><i>Previously, cehsu (Claire Hsu) wrote…</i></summary><blockquote>

I don't imagine a use case at present for including the partials in the templates but not providing them serverside. Maybe we should change this when that is necessary?
</blockquote></details>

Done.

---

*[site/footer.html, line 6 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdH7HrJvgGH12EUGZ9a:-Kdd2AjDjPjqA3y3ho5F:b-896fix) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/site/footer.html#L6)):*
<details><summary><i>Previously, cehsu (Claire Hsu) wrote…</i></summary><blockquote>

Perhaps we should include this for now, for the purposes of eliciting user feedback?
</blockquote></details>

Done.

---

*[site/head.html, line 8 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdHPfmUG9WK2faTMFo7:-Kdd2BQ2zJwAQ0GwvnCT:b-896fix) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/site/head.html#L8)):*
<details><summary><i>Previously, cehsu (Claire Hsu) wrote…</i></summary><blockquote>

Good point, media query now provided.
</blockquote></details>

Done.

---

*[site/login.html, line 7 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdHQqBF4bwQYlV9Nt7Y:-Kdd2C21foGrZPxctsoa:b-896fix) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/site/login.html#L7)):*
<details><summary><i>Previously, cehsu (Claire Hsu) wrote…</i></summary><blockquote>

Good point, removed.
</blockquote></details>

Done.

---

*[site/searchbar.html, line 12 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdHRKV1B0S_F5n0FeCr:-Kdd2CebMtYq1pzdtD9k:b-896fix) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/site/searchbar.html#L12)):*
<details><summary><i>Previously, cehsu (Claire Hsu) wrote…</i></summary><blockquote>

Added as an issue.
</blockquote></details>

Done.

---

*[site/js/helpers.js, line 53 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdI7VYnBNjpEgrFo2ad:-Kdd2DI1HpN5NsZLtmy0:b-896fix) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/site/js/helpers.js#L53)):*
<details><summary><i>Previously, cehsu (Claire Hsu) wrote…</i></summary><blockquote>

?
</blockquote></details>

Done.

---

*[site/js/helpers.js, line 58 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdI6r2pELVBgyHehH6n:-Kdd2Dz4SNlnEhtrUukg:b-896fix) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/site/js/helpers.js#L58)):*
<details><summary><i>Previously, cehsu (Claire Hsu) wrote…</i></summary><blockquote>

Excellent point, additional shortcuts created. Will add caching as a todo for the next CL.
</blockquote></details>

Done.

---

*[site/js/helpers.js, line 125 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdI8i9TtBci2NbjsDtV:-Kdd2Em_rwz23n_2994-:b-896fix) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/site/js/helpers.js#L125)):*
<details><summary><i>Previously, cehsu (Claire Hsu) wrote…</i></summary><blockquote>

Good catch, unnecessary whitespace removed.
</blockquote></details>

Done.

---

*[site/js/helpers.js, line 135 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdI8wGAyqizUAMV7bnN:-Kdd2FSaSsklwT3fTUkP:b-896fix) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/site/js/helpers.js#L135)):*
<details><summary><i>Previously, cehsu (Claire Hsu) wrote…</i></summary><blockquote>

Excellent point, made constant time complexity.
</blockquote></details>

Done.

---

*[site/js/main.js, line 8 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdI9LIBW4U0jxN_Kj7Q:-Kdd2G6wiyyqRoXITqTy:b-896fix) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/site/js/main.js#L8)):*
<details><summary><i>Previously, cehsu (Claire Hsu) wrote…</i></summary><blockquote>

I've added additional description, in the JSDOC block, please let me know if additional explanation would be helpful.
</blockquote></details>

Done.

---

*[site/usermgt/adduser.html, line 13 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdI9eOVMKP3I3PCbp-M:-Kdd2MsQ3E25kpKyeIPs:b-896fix) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/site/usermgt/adduser.html#L13)):*
<details><summary><i>Previously, cehsu (Claire Hsu) wrote…</i></summary><blockquote>

Including Diagnosticmessage instead of ReasonForFailure results in the following message on failed user addition: 
"Can't execute template template: htmlbase:13:10: executing "htmlbase" at <.Diagnosticmessage>: can't evaluate field Diagnosticmessage in type *server.addUserResult"
</blockquote></details>

Done.

---

*[site/usermgt/listusers.html, line 46 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdI9sk1tZ4g_2JIL6AB:-Kdd2NWxaJ3vT86B7Fpl:b-896fix) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/site/usermgt/listusers.html#L46)):*
<details><summary><i>Previously, cehsu (Claire Hsu) wrote…</i></summary><blockquote>

good catch, unnecessary whitespace removed.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103)*
<!-- Sent from Reviewable.io -->

rjkroege commented:
Reviewed 1 of 24 files at r1, 8 of 9 files at r2.
Review status: 22 of 24 files reviewed at latest revision, 5 unresolved discussions.

---

*[server/templating.go, line 39 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdH5rl9tjHAxh3YywWB:-KdfJ06DrGggf-hCfqcY:b-k6e1x0) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/server/templating.go#L39)):*
<details><summary><i>Previously, cehsu (Claire Hsu) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

ok. Aside: a follow up CL (me or you) should do the renaming.

---

*[server/usermgt_test.go, line 30 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdH6V1ytvKjomirUeyB:-KdfJPHve5KkByVc0pUK:bcttv7b) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/server/usermgt_test.go#L30)):*
<details><summary><i>Previously, cehsu (Claire Hsu) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

ok. It's more of an opportunity for code cleanup than any particular issue that needs to be resolved now.

---

*[site/js/helpers.js, line 53 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdI7VYnBNjpEgrFo2ad:-KdfJzSnfIO1hiiF_qNr:b-8puyby) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/site/js/helpers.js#L53)):*
<details><summary><i>Previously, cehsu (Claire Hsu) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

I don't remember what I was going to say. Sorry.

---

*[site/js/main.js, line 8 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdI9LIBW4U0jxN_Kj7Q:-KdfKWz_QpAkd7xzBOjS:bite4ci) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/site/js/main.js#L8)):*
<details><summary><i>Previously, cehsu (Claire Hsu) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

I'd like a little more explanation. And maybe you could line-break the explanation. `preventDefault` and returning true/false from an event handler can have subtle semantics. 

But in the interests of landing sooner rather than later, maybe add explanation later.

---

*[site/usermgt/adduser.html, line 13 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103#-KdI9eOVMKP3I3PCbp-M:-KdfL6g2OHUu_Iqo4A_W:bwelr4o) ([raw file](https://github.com/sfbrigade/sfsbook/blob/9ca2ab11a48cb3c7115b5c3e67d1f995311ce874/site/usermgt/adduser.html#L13)):*
<details><summary><i>Previously, cehsu (Claire Hsu) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

hm. That would be on me I think. I'll attempt to cleanup;

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103)*
<!-- Sent from Reviewable.io -->

rjkroege commented:
Reviewed 1 of 24 files at r1, 1 of 9 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/sfbrigade/sfsbook/103)*
<!-- Sent from Reviewable.io -->
